Services/Services.Application/Services/SpecializationService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using AutoMapper;
+using Services.Application.Models;
+using Services.Application.Models.Enums;
+using Services.Domain.Repositories.Abstractions;
+using Services.Application.Services.Abstractions;
+using Services.Domain.Entities;
+using Services.Domain.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Services.Application.Exceptions;
+
+namespace Services.Application.Services
+{
+    public class SpecializationService : ISpecializationService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public SpecializationService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task Create(CreateSpecializationModel createModel, CancellationToken cancellationToken)
+        {
+            await _unitOfWork.SpecializationRepository.CreateAsync(_mapper.Map<Specialization>(createModel), cancellationToken);
+            await _unitOfWork.SaveChangesAsync(cancellationToken);
+        }
+
+        public async Task Delete(Guid id, CancellationToken cancellationToken)
+        {
+            var transaction = await _unitOfWork.BeginTransactionAsync(cancellationToken: cancellationToken);
```

Comment:
Is transaction object disposable? Add using operator please

Reply:
Fixed